src/main/jbake/content/documentation/bundles/configuration-installer-factory.md

```diff
@@ -70,20 +76,52 @@ The quoted string format is equal to the definition from HTTP 1.1 ([RFC2616](htt
 
 The 1 character type code is one of:
 
-* `T` : simple string
-* `I` : Integer
-* `L` : Long
-* `F` : Float
-* `D` : Double
-* `X` : Byte
-* `S` : Short
-* `C` : Character
-* `B` : Boolean
+* `T` : `String`
+* `I` : `Integer`
+* `L` : `Long`
+* `F` : `Float`
+* `D` : `Double`
+* `X` : `Byte`
+* `S` : `Short`
+* `C` : `Character`
+* `B` : `Boolean`
+
+or for primitives
+
+* `i` : `int`
+* `l` : `long`
+* `f` : `float`
+* `d` : `double`
+* `x` : `byte`
+* `s` : `short`
+* `c` : `char`
+* `b` : `boolean`
 
-For Float and Double types the methods [Float.intBitsToFloat](https://docs.oracle.com/javase/7/docs/api/java/lang/Float.html#intBitsToFloat%28int%29) and [Double.longBitsToDouble](https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#longBitsToDouble%28long%29) are being used to convert the numeric string into the correct type. These methods rely on the IEEE 754 floating-point formats described in [Single-precision floating-point format](https://en.wikipedia.org/wiki/Single-precision_floating-point_format) and [Double-precision floating-point format](https://en.wikipedia.org/wiki/Double-precision_floating-point_format). A more user-friendly format is not yet supported for these types ([SLING-7757](https://issues.apache.org/jira/browse/SLING-7757)).
+For Float and Double types the methods [Float.intBitsToFloat](https://docs.oracle.com/javase/7/docs/api/java/lang/Float.html#intBitsToFloat%28int%29) and [Double.longBitsToDouble](https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#longBitsToDouble%28long%29) are being used to convert the numeric string into the correct type. These methods rely on the IEEE-754 floating-point formats described in [Single-precision floating-point format](https://en.wikipedia.org/wiki/Single-precision_floating-point_format) and [Double-precision floating-point format](https://en.wikipedia.org/wiki/Double-precision_floating-point_format). A more user-friendly format is not yet supported for these types ([SLING-7757](https://issues.apache.org/jira/browse/SLING-7757)).
+
+Multiple values enclosed by `[` and `]` lead to an array while those enclosed by `(` and `)` lead to a Collection in the underlying `java.util.Dictionary` of the `org.osgi.service.cm.Configuration` and vice-versa.
+
+Although both objects and primites are supported, both lead to generating the wrapper objects (and not the primitives) in the underlying `java.util.Dictionary` of the `org.osgi.service.cm.Configuration` during deserialization (even for multivalue entries).
```

Comment:
Um, I don't think this is correct. That will depend on the type given above (upper or lower case) that was the point no?

Reply:
If you look at http://svn.apache.org/viewvc/felix/releases/org.apache.felix.configadmin-1.8.12/src/main/java/org/apache/felix/cm/file/ConfigurationHandler.java?view=markup#l444 you can see that the deserialization (i.e. the part that converts .config file to Dictionary) will always use non-primitives!

Reply:
I don't think that is the case for arrays. They will depend on the type. Because while they first read in using the wrapper (as you point out) they later get set into the array using Array.set() which will automatically unwrap if the array is primitive (which it will be if a lower case type info has been used).